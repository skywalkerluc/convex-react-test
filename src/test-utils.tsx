import { ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';

export const createWrapper = (options?: { router?: { initialEntries?: string[] } }) => {
  const Wrapper = ({ children }: { children: ReactElement }) => (
    <MemoryRouter initialEntries={options?.router?.initialEntries || ['/']}>
      {children}
    </MemoryRouter>
  );
  return Wrapper;
};
