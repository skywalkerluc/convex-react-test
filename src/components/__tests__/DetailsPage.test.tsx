import { render } from '@testing-library/react';

import { createWrapper } from '../../test-utils';
import EstablishmentDetailsPage from '../features/establishments/DetailsPage';

jest.mock('../../hooks/useEstablishmentDetails', () => ({
  useEstablishmentDetails: jest.fn(),
}));

const mockUseEstablishmentDetails = jest.requireMock(
  '../../hooks/useEstablishmentDetails',
).useEstablishmentDetails;

const mockDetails = {
  id: 123,
  businessName: 'Test Restaurant',
  ratingValue: '5',
  ratingDate: '2024-01-01T00:00:00',
  addressLine1: 'Test Street 1',
  addressLine2: 'Test Line 2',
  addressLine3: 'Test Line 3',
  addressLine4: 'Test Line 4',
  postCode: 'TE5 5TP',
  scores: {
    Hygiene: 5,
    Structural: 5,
    ConfidenceInManagement: 5,
  },
};

describe('EstablishmentDetailsPage', () => {
  it('matches snapshot', () => {
    mockUseEstablishmentDetails.mockReturnValue({
      details: mockDetails,
      loading: false,
      error: null,
    });

    const { asFragment } = render(<EstablishmentDetailsPage />, { wrapper: createWrapper() });
    expect(asFragment()).toMatchSnapshot();
  });

  it('shows loading state', () => {
    mockUseEstablishmentDetails.mockReturnValue({
      details: null,
      loading: true,
      error: null,
    });

    const { container } = render(<EstablishmentDetailsPage />, { wrapper: createWrapper() });
    expect(container.querySelector('.spinner')).toBeInTheDocument();
  });

  it('shows error state', () => {
    mockUseEstablishmentDetails.mockReturnValue({
      details: null,
      loading: false,
      error: new Error('Test error'),
    });

    const { getByText } = render(<EstablishmentDetailsPage />, { wrapper: createWrapper() });
    expect(getByText('Error: Test error')).toBeInTheDocument();
  });
});
