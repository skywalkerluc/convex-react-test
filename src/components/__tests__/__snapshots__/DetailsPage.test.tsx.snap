// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EstablishmentDetailsPage matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <button
      class="backButton"
    >
      ‚Üê Back to List
    </button>
    <div
      class="detailsCard"
    >
      <h1
        class="title"
      >
        Test Restaurant
      </h1>
      <div
        class="section"
      >
        <h2
          class="sectionTitle"
        >
          Address
        </h2>
        <div
          class="address"
        >
          <p>
            Test Street 1
          </p>
          <p>
            Test Line 2
          </p>
          <p>
            Test Line 3
          </p>
          <p>
            Test Line 4
          </p>
          <p>
            TE5 5TP
          </p>
        </div>
      </div>
      <div
        class="section"
      >
        <h2
          class="sectionTitle"
        >
          Rating
        </h2>
        <div
          class="ratingBadge"
        >
          5
        </div>
        <p
          class="inspectionDate"
        >
          Inspection Date: 01/01/24
        </p>
      </div>
      <div
        class="section"
      >
        <h2
          class="sectionTitle"
        >
          Scores
        </h2>
        <div
          class="scoresGrid"
        >
          <div
            class="scoreItem"
          >
            <span>
              Hygiene
            </span>
            <span>
              5
            </span>
          </div>
          <div
            class="scoreItem"
          >
            <span>
              Structural
            </span>
            <span>
              5
            </span>
          </div>
          <div
            class="scoreItem"
          >
            <span>
              Confidence
            </span>
            <span>
              5
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
